packages/webdriverio/src/commands/browser/newWindow.ts

```diff
@@ -56,6 +56,10 @@ export default async function newWindow (url, { windowName = 'New Window', windo
     const tabs = await this.getWindowHandles()
     const newTab = tabs.pop()
 
+    if (!newTab) {
```

Comment:
If no new tab is created, it would always return the current tab right? So I think this check would have to be changed to the following.

```js
const originalTabs = await this.getWindowHandles()

await this.execute(newWindowHelper, url, windowName, windowFeatures)

const newTabs = await this.getWindowHandles()

if (newTabs.length !== originalTabs.length + 1) {
    throw new Error('No window handle was found to switch to')
}
```

Reply:
This error check is making sure that there is at least a tab to also make TS happy. I don't want to add any additional checks within this PR.